plugins/sqlfluff-templater-dbt/src/dbt_templater/templater.py

```diff
@@ -1,379 +1,378 @@
-
-        # When using dbt-templater, trailing newlines are ALWAYS REMOVED during
-        # compiling. Unless fixed (like below), this will cause:
-        #    1. L009 linting errors when running "sqlfluff lint foo_bar.sql"
-        #       since the linter will use the compiled code with the newlines
-        #       removed.
-        #    2. "No newline at end of file" warnings in Git/GitHub since
-        #       sqlfluff uses the compiled SQL to write fixes back to the
-        #       source SQL in the dbt model.
-        # The solution is:
-        #    1. Check for trailing newlines before compiling by looking at the
-        #       raw SQL in the source dbt file, store the count of trailing newlines.
-        #    2. Append the count from #1 above to the node.raw_sql and
-        #       compiled_sql objects, both of which have had the trailing
-        #       newlines removed by the dbt-templater.
-        node.raw_sql = node.raw_sql + "\n" * n_trailing_newlines
-        compiled_sql = compiled_sql + "\n" * n_trailing_newlines
-
-        raw_sliced, sliced_file, templated_sql = self.slice_file(
-            node.raw_sql,
-            compiled_sql,
-            config=config,
-        )
-
-        return (
-            TemplatedFile(
-                source_str=node.raw_sql,
-                templated_str=templated_sql,
-                fname=fname,
-                sliced_file=sliced_file,
-                raw_sliced=raw_sliced,
-            ),
-            # No violations returned in this way.
-            [],
-        )
+"""Defines the templaters."""
+
+import os.path
+import logging
+from typing import Optional
+
+from dataclasses import dataclass
+from cached_property import cached_property
+from functools import partial
+
+from sqlfluff.core.errors import SQLTemplaterError, SQLTemplaterSkipFile
+
+from sqlfluff.core.templaters.base import TemplatedFile
+from sqlfluff.core.templaters.jinja import JinjaTemplater
+
+# Instantiate the templater logger
+templater_logger = logging.getLogger("sqlfluff.templater")
+
+
+@dataclass
+class DbtConfigArgs:
+    """Arguments to load dbt runtime config."""
+
+    project_dir: Optional[str] = None
+    profiles_dir: Optional[str] = None
+    profile: Optional[str] = None
+
+
+class DbtTemplater(JinjaTemplater):
+    """A templater using dbt."""
+
+    name = "dbt"
+    sequential_fail_limit = 3
+
+    def __init__(self, **kwargs):
+        self.sqlfluff_config = None
+        self.formatter = None
+        self.project_dir = None
+        self.profiles_dir = None
+        self.working_dir = os.getcwd()
+        self._sequential_fails = 0
+        super().__init__(**kwargs)
+
+    def config_pairs(self):
+        """Returns info about the given templater for output by the cli."""
+        return [("templater", self.name), ("dbt", self.dbt_version)]
+
+    @cached_property
+    def dbt_version(self):
+        """Gets the dbt version."""
+        from dbt.version import get_installed_version

```

Comment:
Now that it's a plugin, would it be safe to assume that dbt is installed and (thus) move the dbt imports to the top of the file?

Reply:
yes good point